### ISSUE TITLE
Authentication issues when using IPv6 via robbertkl/ipv6nat

### ISSUE THREAD

**Issue by cooperaj** [Mailu/Mailu#827]
I'm using robbertkl/ipv6nat to simplify my IPv6 management and it's causing issues with auth due to the new $SUBNET requirement for things. 

TLDR. I believe that IPv6 causes issues for containers that communicate over IPv6 since the authentication is based on the IPv4 $SUBNET property.

Using robbertkl/ipv6nat lets me treat IPv6 on my server in an identical fashion to my normal IPv4 services. It does standard NAT using the iptables firewall so that opening ports on a container works for both standards without changes.

The issue is that, on an apparent random basis (I think) the internal traffic uses the ULA local addresses to communicate amongst containers and as this does not match the $SUBNET things sometimes don't work. The most noticeable service with this issue is the Rspamd UI which spews near constant error messages, which when viewed in the inspector, are 403 errors. 

My (abbreviated) compose file is:

```
version: '2.1'

services:

  front:
    image: mailu/nginx:$VERSION
    restart: always
    env_file: .env
    logging:
      driver: $LOG_DRIVER
    ports:
    - "80:80"
    - "443:443"
    - "110:110"
    - "143:143"
    - "993:993"
    - "995:995"
    - "25:25"
    - "465:465"
    - "587:587"
    volumes:
      - "$ROOT/certs:/certs"
      - "$ROOT/overrides/nginx:/overrides"

... other services ...

networks:
  default:
    enable_ipv6: true
    ipam:
      driver: default
      config:
        - subnet: $SUBNET
        - subnet: $SUBNET6
```

As you can see I've added some IPv6 config to the network which is needed for the ipv6nat container to be able to route to the services.

The reason I setup the system this way (and not even because I tend to always do it) is because without doing this the standard docker-compose file you recommend would have my IPv6 services be inaccessible. I was not able to have any service connect to them. This is on a standard Ubuntu 18.04 VM with the latest vendor Docker installed and the UFW firewall running. Using the ipv6nat tool allows me to keep the firewall without changes as docker then manages the IPv6 the same way as IPv4.

**Comment by muhlemmer** [comment]
I will investigate, probably after the weekend.

**Comment by muhlemmer** [comment]
Unfortunately this seems due to a flaky docker resolver. When using the settings that you've proposed, I was able to reproduce:
```
imap_1       | ERROR:asyncio:_GatheringFuture exception was never retrieved
imap_1       | future: <_GatheringFuture finished exception=ClientConnectorError(-2, 'Name does not resolve')>
imap_1       | Traceback (most recent call last):
imap_1       |   File "/usr/lib/python3.6/site-packages/aiohttp/connector.py", line 955, in _create_direct_connection
imap_1       |     traces=traces), loop=self._loop)
imap_1       |   File "/usr/lib/python3.6/site-packages/aiohttp/connector.py", line 825, in _resolve_host
imap_1       |     self._resolver.resolve(host, port, family=self._family)
imap_1       |   File "/usr/lib/python3.6/site-packages/aiohttp/resolver.py", line 30, in resolve
imap_1       |     host, port, type=socket.SOCK_STREAM, family=family)
imap_1       |   File "/usr/lib/python3.6/concurrent/futures/thread.py", line 56, in run
imap_1       |     result = self.fn(*self.args, **self.kwargs)
imap_1       |   File "/usr/lib/python3.6/socket.py", line 745, in getaddrinfo
imap_1       |     for res in _socket.getaddrinfo(host, port, family, type, proto, flags):
imap_1       | socket.gaierror: [Errno -2] Name does not resolve
imap_1       | 
imap_1       | The above exception was the direct cause of the following exception:
imap_1       | 
imap_1       | Traceback (most recent call last):
imap_1       |   File "/usr/lib/python3.6/site-packages/podop/dovecot.py", line 119, in process_commit
imap_1       |     ns=(self.user if key_type == "priv" else None)
imap_1       |   File "/usr/lib/python3.6/site-packages/podop/table.py", line 45, in set
imap_1       |     await session.post(uri, json=value)
imap_1       |   File "/usr/lib/python3.6/site-packages/aiohttp/client.py", line 476, in _request
imap_1       |     timeout=real_timeout
imap_1       |   File "/usr/lib/python3.6/site-packages/aiohttp/connector.py", line 522, in connect
imap_1       |     proto = await self._create_connection(req, traces, timeout)
imap_1       |   File "/usr/lib/python3.6/site-packages/aiohttp/connector.py", line 854, in _create_connection
imap_1       |     req, traces, timeout)
imap_1       |   File "/usr/lib/python3.6/site-packages/aiohttp/connector.py", line 959, in _create_direct_connection
imap_1       |     raise ClientConnectorError(req.connection_key, exc) from exc
imap_1       | aiohttp.client_exceptions.ClientConnectorError: Cannot connect to host admin:80 ssl:None [Name does not resolve]
```
This only happens from time to time. So I decided to run `docker-compose exec front nslookup admin` repeatably and found that there is no result from time to time.  I also tried this on an IPv4-only network and things work fine.

Yes I agree that the IPv6 bind addresses are pretty useless until Docker finally fixes the whole IPv6 stack. I'm sorry, but I'm afraid there's not much we can do here. (Filing a IPv6 related bug to Docker is a hopeless cause at the moment, but feel free to try)
